driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -120,31 +124,33 @@ public void run() {
 
         Executor initExecutor = manager.cluster.manager.configuration.getPoolingOptions().getInitializationExecutor();
 
-        ListenableFuture<List<Void>> allConnectionsFuture = Futures.allAsList(connectionFutures);
+        ListenableFuture<List<Void>> allConnectionsFuture = Futures.successfulAsList(connectionFutures);
 
         final SettableFuture<Void> initFuture = SettableFuture.create();
-        Futures.addCallback(allConnectionsFuture, new FutureCallback<List<Void>>() {
+        Futures.addCallback(allConnectionsFuture, new MoreFutures.SuccessCallback<List<Void>>() {
             @Override
             public void onSuccess(List<Void> l) {
+                // Some of the connections might have failed, keep only the successful ones
+                ListIterator<Connection> it = connections.listIterator();
```

Comment:
It looks like this *shouldn't* be a problem if we get a connection timeout because the Futures.withFallback will cover channelReadyFuture failing, but thought I'd point out that scenario as well.